File: messenger-ui/src/service/MessageService.js
import {$authApi} from "../http";

export default class MessageService {

    static async fetchMessages(chatId, page = 0, size = 10) {
        return $authApi.get(`/private-chats/${chatId}/messages?page=${page}&size=${size}`)
    }

    static async updateMessage(updatedMessage) {
        return $authApi.patch(`/private-chat-messages/${updatedMessage.id}`, updatedMessage)
    }

    static async createMessage(chatId, content) {
        return $authApi.post(`/private-chats/${chatId}/messages`, {content: content})
    }

    static async deleteMessage(messageId) {
        return $authApi.delete(`/private-chat-messages/${messageId}`)
    }
}

File: messenger-server/src/main/java/io/github/artemfedorov2004/messengerserver/service/MessageService.java
package io.github.artemfedorov2004.messengerserver.service;

import io.github.artemfedorov2004.messengerserver.controller.payload.NewMessagePayload;
import io.github.artemfedorov2004.messengerserver.controller.payload.UpdateMessagePayload;
import io.github.artemfedorov2004.messengerserver.entity.Chat;
import io.github.artemfedorov2004.messengerserver.entity.Message;
import io.github.artemfedorov2004.messengerserver.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface MessageService {

    Message createMessage(Chat chat, User sender, NewMessagePayload payload);

    Page<Message> getByChatId(Long chatId, Pageable pageable);

    Message updateMessage(Long messageId, UpdateMessagePayload payload);

    Message getMessage(Long messageId);

    void deleteMessage(Long id);

    boolean canGetMessage(Long messageId, User principal);

    boolean canEditMessage(Long messageId, User principal);

    boolean canDeleteMessage(Long messageId, User principal);

    Optional<Message> findLastMessageByChatId(Long chatId);
}
